dao/src/main/kotlin/tables/ScanResultsTable.kt

```diff
@@ -91,20 +91,20 @@ class ScanResultDao(id: EntityID<Long>) : LongEntity(id) {
         val provenance = when {
             artifactUrl != null -> ArtifactProvenance(
                 sourceArtifact = RemoteArtifact(
-                    url = artifactUrl!!,
+                    url = checkNotNull(artifactUrl),
                     hashValue = artifactHash.orEmpty(),
                     hashAlgorithm = artifactHashAlgorithm.orEmpty()
                 )
             )
 
             vcsUrl != null -> RepositoryProvenance(
                 vcsInfo = VcsInfo(
-                    type = RepositoryType.forName(vcsType!!),
-                    url = vcsUrl!!,
-                    revision = vcsRevision!!,
+                    type = RepositoryType.forName(vcsType.orEmpty()),
+                    url = checkNotNull(vcsUrl),
+                    revision = vcsRevision.orEmpty(),
                     path = ""
                 ),
-                resolvedRevision = vcsRevision!!
+                resolvedRevision = vcsRevision.orEmpty()
```

Comment:
Better use `checkNotNull` for all values, if one of them is null it is an indication of a bug and failing early helps with disovering it.